patches/CreatePBP.cmake

```diff
@@ -0,0 +1,55 @@
+# File defining macro outputting PSP-specific EBOOT.PBP out of passed executable target.
+# Copyright 2020 - Daniel 'dbeef' Zalega
+#
+# Args:
+# target - defined by an add_executable call before calling create_pbp_file
+# icon_path - path to .png file, 144x82, pass "NULL" to use default
+# background_path - path to .png file, 480x272, pass "NULL" to use default
+# preview_path - path to .png file, 480x272, pass "NULL" to use default
+
+cmake_minimum_required(VERSION 3.10)
+
+macro(create_pbp_file target icon_path background_path preview_path)
+
+    add_custom_command(TARGET ${target} POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E make_directory
+            $<TARGET_FILE_DIR:${target}>/psp_artifact
+            COMMENT "Creating psp_artifact directory."
+            )
+
+    add_custom_command(TARGET ${target} POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E copy_if_different
+            $<TARGET_FILE:${target}>
+            $<TARGET_FILE_DIR:${target}>/psp_artifact
+            COMMENT "Copying ELF to psp_arfitact directory."
+            )
+
+    add_custom_command(
+            TARGET ${target}
+            POST_BUILD COMMAND
+            "psp-fixup-imports" "$<TARGET_FILE_DIR:${target}>/psp_artifact/${target}"
+            COMMENT "Calling psp-fixup-imports"
+    )
+
+    add_custom_command(
+            TARGET ${target}
+            POST_BUILD COMMAND
+            "mksfoex" "-d" "MEMSIZE=1" "$<TARGET_FILE_DIR:${target}>/psp_artifact/${target}" "PARAM.SFO"
```

Comment:
We should make `MEMSIZE` configurable, as we are currently omitting psp-1000 support.
There is support for optional parameters in cmake functions.

Reply:
I tested `MEMSIZE=1` on my PSP-1000 and it is simply ignored, so I would leave it as it is.
Same goes for PPSSPP, when ran in PSP-1000 mode it logs:

```
42:54:195 Core/PSPLoaders.cpp:70 W[LOADER]: Game requested full PSP-2000 memory access, ignoring in PSP-1000 mode
```